plugins/algoltek-usb/fu-algoltek-usb-device.c

```diff
@@ -0,0 +1,514 @@
+/*
+ * Copyright (C) 2023 Algoltek, Inc.
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include "fu-algoltek-usb-common.h"
+#include "fu-algoltek-usb-device.h"
+#include "fu-algoltek-usb-firmware.h"
+#include "fu-algoltek-usb-struct.h"
+
+struct _FuAlgoltekUsbDevice {
+	FuUsbDevice parent_instance;
+};
+
+G_DEFINE_TYPE(FuAlgoltekUsbDevice, fu_algoltek_usb_device, FU_TYPE_USB_DEVICE)
+
+static gboolean
+algoltek_device_ctrl_transfer(FuAlgoltekUsbDevice *self,
+			      GUsbDeviceDirection direction,
+			      guint8 algoltek_cmd,
+			      guint16 value,
+			      guint16 index,
+			      GByteArray *transfer_pkt,
```

Comment:
I think `buf` is the usual name for this.

Reply:
Updated!